changelog/fragments/align-kb.yaml

```diff
@@ -0,0 +1,124 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      For Go-based operators, migrate your project to use the project version stable.
+    kind: "change"
+    breaking: true
```

Comment:
I'm wondering whether we can auto-migrate from `3-alpha` to `3`. All the info needed can be found in the `3-alpha` project config right @Adirio @camilamacedo86?

Reply:
We could migrate some info, like the domain, path and plurals as they can only be the defauls pre v3.0.0, but whether the controller, resource or webhooks were scaffolded, that we can't auto-migrate.

Reply:
HI @estroz, 

The motivations to add to the PROJECT file the fields are for in the future we be able to try to tooling things. However, has no tooling from v3-alpha to v3 stable and I do not think that this achievable.

Reply:
>the controller, resource or webhooks were scaffolded, that we can't auto-migrate.

@Adirio detecting default paths to see if these were scaffolded would work, and if they aren't found then print a message like "if you created a resource, controller, or webhook, please ensure that the appropriate field is set to true in your PROJECT file. See this migration guide for more information: https://master.book.kubebuilder.io/migration/plugin/plugins.html"

>I do not think that this achievable.

@camilamacedo86 why not? This could be a pre-command hook that runs before any plugin-based command of `operator-sdk`, only if `3-alpha` is detected. It would be removed in v2.0.0.

Reply:
Hi @estroz, 

The motivations for v3 stable layout is to have the data used to do the scaffold. 
Therefore, by having this data would be possible to perform toolings and helpers in the future. 
However, I do not see how could  be achievable an auto-migrate tool from  v3-alpha to v3

Anyway, since SDK releases 1.3.0 and 1.4.0 added the go/v3 as a default plugin. I understand that it is an impact that we were aware of it already. Am I right?

Reply:
I guess that would be doable, yes. Would work for most of the projects and would require just a frw amount to do manual changes, which would be nice.

Reply:
@estroz, @adirio, 
 
My idea before was:
- Go/v2 uses project version 2 or v3-alpha ( with the resources.api and resources.webhook correct layout as agreed )
- And go/v3 uses v3 project version stable
It would mean that who is using v2 would not be impacted. And I think it is achievable yet. 

I am not sure about the hook tool suggested. Any customization could make it does not work properly.  
c/c @jmrodri

Reply:
Hi @estroz, @Adirio, 

I do not think that would be a good approach to make the tool try to re-write the PROJECT file automatically because it is impossible to ensure that the values will be accurate. 

And then, KB never released a stable version with 3-alpha. 

Then, I understand that if someone tries to use the tool after with 3-alpha it will fail. So, we could, in this case, raise an issue with the link for the migration guide after it is merged. See: https://github.com/kubernetes-sigs/kubebuilder/pull/1916/files

c/c @jmrodri

Reply:
It shows solved.